### DISCUSSION TITLE
Can't get text from the pdf

### DISCUSSION THREAD

**Discussion by hemilparmar** [pymupdf/PyMuPDF#2090] [Looking for help] [answered]
when i parse this pdf i can not get the content of pdf
[test.pdf](https://github.com/pymupdf/PyMuPDF/files/10103665/test.pdf)

Code:
with fitz.open("test.pdf") as f:
    for p in f:
        print("\n\n")
        print(p.get_text(sort=True))

**Comment by hemilparmar** [comment]
please solve above issue.

**Comment by JorjMcKie** [comment, marked as answer]
> please solve above issue.

Sorry, impossible! File contains no text. Except for the 2 pages containing the table of contents.

Please **_always_** in those cases try whether your PDF viewer allows you to select (as if for copy / paste) stuff that appears to be text. If this is not possible, then you know you have no text there!

**Reply by hemilparmar** [reply]
Okay,
[test2.pdf](https://github.com/pymupdf/PyMuPDF/files/10103838/test2.pdf)
can you see above file ?
this file gives random symbols instead of text.

**Reply by JorjMcKie** [reply]
What does Adobe Acrobat give you when copy/pasting (e.g. page 1) to a text file?

You will see: junk as well!
Same with any PDF viewer!

**Reply by hemilparmar** [reply]
I get junk in text file. 
What is the reason behind it ? and is there any way yo extract text ?

**Reply by JorjMcKie** [reply]
> I get junk in text file. What is the reason behind it ? and is there any way yo extract text ?

Aha!
No way out except OCR. Either use PyMuPDF's integrated feature or some other package.
The reason behind this only knows the PDF creator. He obviously was against text extraction.

**Reply by hemilparmar** [reply]
Okay, Thanks
